addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -660,8 +660,20 @@ odoo.define('website.s_website_form', function (require) {
          * Calculates the visibility for each field with conditional visibility
          */
         _updateFieldsVisibility() {
+            let anyFieldVisibilityUpdated = false;
             for (const [fieldEl, visibilityFunction] of this._visibilityFunctionByFieldEl.entries()) {
-                this._updateFieldVisibility(fieldEl, visibilityFunction());
+                const wasVisible = !fieldEl.closest(".s_website_form_field")
+                    .classList.contains("d-none");
+                const isVisible = !!visibilityFunction();
+                this._updateFieldVisibility(fieldEl, isVisible);
+                anyFieldVisibilityUpdated = wasVisible !== isVisible || anyFieldVisibilityUpdated;
```

Comment:
I wanted to suggest:
```js
anyFieldVisibilityUpdated |= wasVisible !== isVisible;
```
but this operator seems to only be used once in the JS codebase (except libs). So do as you prefer.

Reply:
I never used this operator, but it's well suited for this case and its conciseness is nice. :) Thanks